trino/trino_test.go

```diff
@@ -765,6 +765,106 @@ func TestQueryColumns(t *testing.T) {
 	assert.Equal(t, actualTypes, expectedTypes)
 }
 
+func TestParametricDateTime(t *testing.T) {
+	c := &Config{
+		ServerURI:         *integrationServerFlag,
+		SessionProperties: map[string]string{"query_priority": "1"},
+	}
+
+	dsn, err := c.FormatDSN()
+	require.NoError(t, err)
+
+	db, err := sql.Open("trino", dsn)
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		assert.NoError(t, db.Close())
+	})
+
+	rows, err := db.Query(`SELECT
+  cast(current_time as time(6)) AS timep,
+  cast(current_time as time(6) with time zone) AS timeptz,
+  cast(current_time as timestamp(6)) AS tsp,
+  cast(current_time as timestamp(6) with time zone) AS tsptz`,
+		sql.Named("X-Trino-Client-Capabilities", "PARAMETRIC_DATETIME"))
+	require.NoError(t, err, "Failed executing query")
+	assert.NotNil(t, rows)
+
+	columns, err := rows.Columns()
+	require.NoError(t, err, "Failed reading result columns")
+
+	assert.Equal(t, 4, len(columns), "Expected 4 result column")
+	expectedNames := []string{
+		"timep",
+		"timeptz",
+		"tsp",
+		"tsptz",
+	}
+	assert.Equal(t, expectedNames, columns)
+
+	columnTypes, err := rows.ColumnTypes()
+	require.NoError(t, err, "Failed reading result column types")
+
+	assert.Equal(t, 4, len(columnTypes), "Expected 4 result column type")
+
+	type columnType struct {
+		typeName  string
+		hasScale  bool
+		precision int64
+		scale     int64
+		hasLength bool
+		length    int64
+		scanType  reflect.Type
+	}
+	expectedTypes := []columnType{
+		{
+			"TIME",
+			false,
+			0,
+			0,
+			false,
+			0,
+			reflect.TypeOf(sql.NullTime{}),
+		},
+		{
+			"TIME WITH TIME ZONE",
+			false,
+			0,
+			0,
+			false,
+			0,
+			reflect.TypeOf(sql.NullTime{}),
+		},
+		{
+			"TIMESTAMP",
+			false,
+			0,
+			0,
+			false,
+			0,
+			reflect.TypeOf(sql.NullTime{}),
+		},
+		{
+			"TIMESTAMP WITH TIME ZONE",
+			false,
+			0,
```

Comment:
Shouldn't this be non-zero?

Reply:
Added support to capture timestamp and time column precision.